### PR TITLE
47 program register

### LINKED ISSUE #47: Player should be able to program their registers
* `N` cards to choose from (based on health)
*  Be able to drag and drop on control panel cards ([see here](https://github.com/libgdx/libgdx/blob/a4805d6a017b80622d6bfdd3a791352257a3c539/tests/gdx-tests/src/com/badlogic/gdx/tests/DragAndDropTest.java))

### PR DESCRIPTION
**Pull request by kh498**
Fixes #47

### CODE REVIEW

File: src/main/java/no/uib/inf112/core/ui/CardContainer.java [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+        handCard = new CardSlot[Player.MAX_PLAYER_CARDS];
+        drawnCard = new CardSlot[Player.MAX_DRAW_CARDS];
+    }
+
+    /**
+     * Draw cards for the players drawn cards and removes any non-disabled cards from the players hand
+     */
+    public void draw() {
+//        System.out.println("handCard = " + Arrays.toString(handCard));
```
Comment (DanielBerge) [inline src/main/java/no/uib/inf112/core/ui/CardContainer.java]: Remove these if they where only for debugging

File: src/main/java/no/uib/inf112/core/ui/CardContainer.java [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+                drawnCard[i].getColor().a = 0.70f;
+            } else {
+                drawnCard[i].setCard(draw[i]);
+            }
+        }
+
+    }
+
+
+    //FIXME probably broken
```
Comment (DanielBerge) [inline src/main/java/no/uib/inf112/core/ui/CardContainer.java]: Make an issue?

File: src/main/java/no/uib/inf112/core/ui/CardContainer.java [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+                for (CardSlot drawnCard : drawnCard) {
+                    if (drawnCard.getCard() == null) {
+                        drawnCard.setCard(handCard.getCard());
+                    }
+                }
+            }
+        }
+
+        //TODO assert handCards is empty/disabled
```
Comment (DanielBerge) [inline src/main/java/no/uib/inf112/core/ui/CardContainer.java]: There are a lot of todos in this class not referenced to issues

File: src/main/java/no/uib/inf112/core/ui/actors/cards/CardSlot.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        this.type = type;
+        this.container = container;
+        slotId = id;
+
+        dad.addSource(new SlotSource(this));
+        dad.addTarget(new SlotTarget(this));
+        addListener(new TooltipListener(this));
+    }
+
+    //TODO test if the card is disabled when it should be
```
Comment (DanielBerge) [inline src/main/java/no/uib/inf112/core/ui/actors/cards/CardSlot.java]: Issue?

File: src/main/java/no/uib/inf112/core/ui/actors/cards/SlotSource.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        final CardSlot payloadSlot = (CardSlot) payload.getObject();
+        if (target != null) {
+            final CardSlot targetSlot = (CardSlot) target.getActor();
+
+            if (targetSlot.isDisabled()) {
+                //do not allow dropping on disabled slots (both drawn and hand)
+                return;
+            }
+
+//            System.out.println("Dropping card on card " + targetSlot.getSlotId() + " from " + targetSlot.getSlotType());
```
Comment (DanielBerge) [inline src/main/java/no/uib/inf112/core/ui/actors/cards/SlotSource.java]: Should this still be here?

**Comment by kh498** [comment]
All done @DanielBerge

**Pull request merged**